prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt

```diff
@@ -209,38 +201,29 @@ class ProcessFhirCommands : CliktCommand(
      * look like.
      * @return a FHIR bundle and the parsed HL7 input that represents the data in the one HL7 message
      */
-    private fun convertHl7ToFhir(hl7String: String, actionLogger: ActionLogger): Pair<Bundle, Message> {
+    private fun convertHl7ToFhir(hl7String: String): Pair<Bundle, Message> {
         val hasFiveEncodingChars = hl7MessageHasFiveEncodingChars(hl7String)
         // Some HL7 2.5.1 implementations have adopted the truncation character # that was added in 2.7
         // However, the library used to encode the HL7 message throws an error it there are more than 4 encoding
         // characters, so this work around exists for that scenario
         val stringToEncode = hl7String.replace("MSH|^~\\&#|", "MSH|^~\\&|")
-        val messages = HL7Reader(actionLogger).getMessages(stringToEncode)
-        if (messages.isEmpty()) throw CliktError("No HL7 messages were read.")
-        val message = if (messages.size > 1) {
-            if (hl7ItemIndex == null) {
-                throw CliktError("Only one HL7 message can be converted. Use the --hl7-msg-index.")
-            } else if (hl7ItemIndex!! < 0 || hl7ItemIndex!! >= messages.size) {
-                throw CliktError("Invalid HL7 message index. Must be a number 0 to ${messages.size - 1}.")
-            } else {
-                messages[hl7ItemIndex!!]
-            }
-        } else {
-            messages[0]
-        }
+        val hl7message = HL7Reader.parseHL7Message(
```

Comment:
I ended up switching this out because the translation tests now need to work this way too so I wanted to keep them consistent